geode-protobuf-client/src/main/java/org/apache/geode/internal/cache/client/protobuf/Region.java

```diff
@@ -0,0 +1,204 @@
+
+    final InputStream inputStream = socket.getInputStream();
+    final RegionAPI.GetAllResponse getAllResponse =
+        ClientProtocol.Message.parseDelimitedFrom(inputStream).getResponse().getGetAllResponse();
+    for (BasicTypes.Entry entry : getAllResponse.getEntriesList()) {
+      // TODO: How does one get a java.lang.Object out of
+      // org.apache.geode.internal.protocol.protobuf.v1.EncodedValue?
+      // TODO values.put((K) entry.getKey().getStringResult(), (V)
+      // entry.getValue().getStringResult());
+    }
+
+    return values;
+  }
+
+  public void put(K key, V value) throws Exception {
+    final OutputStream outputStream = socket.getOutputStream();
+    ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder()
+            .setPutRequest(RegionAPI.PutRequest.newBuilder().setRegionName(name)
+                .setEntry(encodeEntry(key.toString(), value))))
+        .build().writeDelimitedTo(outputStream);
+
+    final InputStream inputStream = socket.getInputStream();
+    ClientProtocol.Message.parseDelimitedFrom(inputStream).getResponse().getPutResponse();
+  }
+
+  public void putAll(Map<K, V> values) throws Exception {
+    final OutputStream outputStream = socket.getOutputStream();
+    RegionAPI.PutAllRequest.Builder putAllRequest = RegionAPI.PutAllRequest.newBuilder();
+    putAllRequest.setRegionName(name);
+    for (K key : values.keySet()) {
+      putAllRequest.addEntry(encodeEntry(key, values.get(key)));
+    }
+    ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder().setPutAllRequest(putAllRequest)).build()
+        .writeDelimitedTo(outputStream);
+
+    final InputStream inputStream = socket.getInputStream();
+    final RegionAPI.PutAllResponse putAllResponse =
+        ClientProtocol.Message.parseDelimitedFrom(inputStream).getResponse().getPutAllResponse();
+    if (0 < putAllResponse.getFailedKeysCount()) {
+      StringBuilder builder = new StringBuilder();
+      for (BasicTypes.KeyedError keyedError : putAllResponse.getFailedKeysList()) {
+        if (0 < builder.length()) {
+          builder.append(", ");
+        }
+        builder.append(decodeValue(keyedError.getKey()).toString());
+      }
+      throw new Exception("Unable to put the following keys: " + builder.toString());
+    }
+  }
+
+  public void remove(K key) throws Exception {
+    final OutputStream outputStream = socket.getOutputStream();
+    ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder().setRemoveRequest(RegionAPI.RemoveRequest
+            .newBuilder().setRegionName(name).setKey(encodeValue(key.toString()))))
+        .build().writeDelimitedTo(outputStream);
+
+    final InputStream inputStream = socket.getInputStream();
+    ClientProtocol.Message.parseDelimitedFrom(inputStream).getResponse().getRemoveResponse();
+  }
+
+  public void removeAll(Collection<K> keys) throws Exception {
```

Comment:
This call will need to be removed as we never implemented this on the server side.